lib/common/patchset.c

```diff
@@ -1034,6 +1034,47 @@ apply_v2_patchset(xmlNode *xml, const xmlNode *patchset)
     return rc;
 }
 
+/*!
+ * \internal
+ * \brief Check whether the digest after applying an XML patchset is as expected
+ *
+ * \param[in]     input     XML before applying patchset
+ * \param[in,out] result    XML after applying patchset
+ * \param[in]     patchset  XML patchset
+ * \param[in]     expected  Expected digest
+ *
+ * \note This macro assumes that an \c int variable called \c rc is already in
+ *       scope.
+ * \note This is a macro (rather than a function) to facilitate trace logging.
+ *       The caller makes a copy of \p input before applying the patchset only
+ *       if \p input might be saved to a file with tracing enabled.
+ */
+#define compare_digests(input, result, patchset, expected) do {             \
```

Comment:
I'm not tied to macro-izing digest comparison or (in the next commit) trace logging. I did this when I thought I'd be using these blocks twice. That's no longer the case.

However, as annoying as macros are, these do make the caller look cleaner.